### PR TITLE
[master] Add check if user directory exists in guest_get_power_state

### PR DESCRIPTION
**Pull request by SeanHQF**
Currently guest_get_power_state returns shutdown if the guest userid
exists in DB but does not exist in user directory (e.g. it has been
deleted through smcli directly), and this is wrong which could cause
confusion and unexpected result.
Also because if the userid does not exist in DB, it returns 404.
Hence, the fix returns 404 to caller in this case.

Signed-off-by: QingFeng Hao <haoqf@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by SeanHQF** [comment]
@bjhuangr @jichenjc Please help review, thanks
@trenxiaof

**Comment by SeanHQF** [comment]
It has been local tested and works as expected:
2021-06-24 07:28:43.003 700090 ERROR nova.virt.zvm.utils [-] zVM Cloud Connector request guest_get_power_state failed with parameters: ('hqf00054',) {} .  Results: {'overallRC': 404, 'modID': 10, 'rc': 404, 'rs': 3, 'errmsg': "Guest 'HQF00054' does not exist in directory although it is in DB. The guest could have been deleted out of z/VM Cloud Connector.", 'output': ''}
2021-06-24 07:28:43.003 700090 WARNING nova.virt.zvm.hypervisor [-] Failed to get power state due to nonexistent instance: hqf00054: nova.exception.ZVMConnectorError: zVM Cloud Connector request failed: {'overallRC': 404, 'modID': 10, 'rc': 404, 'rs': 3, 'errmsg': "Guest 'HQF00054' does not exist in directory although it is in DB. The guest could have been deleted out of z/VM Cloud Connector.", 'output': ''}
2021-06-24 07:28:43.003 700090 INFO nova.compute.manager [-] ICIC: instance not found

File: zvmsdk/api.py [hunk 1]
```diff
@@ -200,6 +200,12 @@ def guest_unpause(self, userid):
     @check_guest_exist()
     def guest_get_power_state(self, userid):
         """Returns power state."""
+        if not zvmutils.check_userid_exist(userid.upper()):
```
Comment (jichenjc) [inline zvmsdk/api.py:203]: has this been verified through SSI cluster? the function seems not tetsed on SSI before?

```
def check_userid_exist(userid, needLogon=False):
    """The successful output is: FBA0004  - DSC
    The successful output for device is (vmcp q 0100):
    DASD 0100 3390 IAS106 R/W      29128 CYL ON DASD  1356 SUBCHANNEL = 0003

    Errors are:
    HCPCQV003E Invalid option - XXXXX
    HCPQVD040E Device XXXX does not exist
    HCPCFC026E Operand missing or invalid
    HCPCQU045E XXXXX not logged on

    Success msgs:
    HCPCQU361E LOGOFF/FORCE pending for user xxxxxx
    """

```

Comment (SeanHQF) [reply zvmsdk/api.py:203]: good finding. I tested on ssi env and found that the cmd returns successfully on all nodes even if VM is created on one of them.
```
[root@ssi001 site-packages]# vmcp q SSI2000a
SSI2000A - DSC

[root@ssi002 ~]# vmcp q SSI2000a
SSI2000A - SSI
```
The result is different for non-ssi env.
How to deal with this?

Comment (SeanHQF) [reply zvmsdk/api.py:203]: Check the result if it contains SSI? if it contains, then regarding it not on this node?

Comment (jichenjc) [reply zvmsdk/api.py:203]: this seems not related ,we can use follow up to fix the problem

@bjhuangr do you have comments on the PR itself?

**Comment by SeanHQF** [comment]
Pls help merge if no further comments, thx

**Pull request merged**